official/keras_application_models/model_callbacks.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Callbacks for Keras built-in application models.
+
+Note that, Keras Callbacks provide two levels (batch and epoch level) to
+record the internal states and statistics during model training. So in the
+following customized callbacks, we also provide two levels for benchmark:
+`batch_based` and `epoch_based`. Users can specify the `--benchmark_level` to
+choose benchmark level. This is similar to the `every_n_seconds` and
+`every_n_steps` options in tensorflow Estimator hooks.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import time
+
+# pylint: disable=g-bad-import-order
+import tensorflow as tf
+# pylint: enable=g-bad-import-order
+
+from official.utils.logs import logger
+
+CALLBACKS = ["examplespersecondcallback", "loggingmetriccallback"]
+
+_METRICS_TO_LOG = {
+    "loss": "train_loss",
+    "acc": "train_accuracy",
+    "val_loss": "loss",
+    "val_acc": "accuracy"
+}
+
+
+class ExamplesPerSecondCallback(tf.keras.callbacks.Callback):
+  """ExamplesPerSecond callback.
+
+  This callback records the average examples per second during training.
+  """
+
+  def __init__(self, batch_size=None, epoch_size=None, batch_based=None,
```

Comment:
I think having both boolean param and also int param is bit redundant. If user specified batch_size, I think its clear that they want to measure based on batch.

Reply:
You mean when users specify batch_size in command line, we should use batch_based benchmark? as we have default batch_size for the model. When users use callbacks, they just specify names. Maybe we should accept more args like hooks?